vadim.martsun/task_6/src/main/java/com/hillel/spring/mvc/dao/userRepository/UserRepositoryImpl.java

```diff
@@ -0,0 +1,88 @@
+package com.hillel.spring.mvc.dao.userRepository;
+
+import com.hillel.spring.mvc.dao.accountRepository.AccountRepository;
+import com.hillel.spring.mvc.model.Account;
+import com.hillel.spring.mvc.model.User;
+import com.hillel.spring.mvc.model.mappers.userMapper.UserMapper;
+import com.hillel.spring.mvc.model.requests.UserRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Repository
+public class UserRepositoryImpl implements UserRepository {
+
+    private Map<Integer, User> userMap = new ConcurrentHashMap<>();
+    private int userId = 0;
+
+    @Autowired
+    private UserMapper userMapper;
+
+    @Autowired
+    private AccountRepository accountRepository;
+
+    @Override
+    public List<User> getAllUsers() {
+        return new ArrayList<>(userMap.values());
+    }
+
+    @Override
+    public User getUserById(int id) {
+        return userMap.getOrDefault(id, null);
+    }
+
+    @Override
+    public boolean update(int id, UserRequest updatedUser) {
+        return update(id, userMapper.getUser(updatedUser));
+    }
+
+    @Override
+    public boolean update(int id, User updatedUser) {
+        User previous = userMap.get(id);
+        if(previous == null) return false;
+
+        for(Map.Entry<Integer, Account> entry : accountRepository.getMap().entrySet()){
```

Comment:
It's bad style  and incorrect usage of hashmap. Don't interate hashmap to find something by value.
You should access hashmap by key.

If you need - better to create one more map like
private Map<User, Account> userAccountMap = new ConcurrentHashMap<>();